### DISCUSSION TITLE
Run examples

### DISCUSSION THREAD

**Discussion by zoechi** [RibirX/Ribir#259] [General]
Is there a way to just run the examples from the `ribir/examples` directory in the repo?
Is there a reason why they are "hidden" in the "ribir" directory instead of being their own crate at the top level of the directory hierarchy?

**Comment by M-Adoo** [comment]
- Just run `cargo run --example counter` in the workspace. 
- 'ribir' is a library that reexports all the required stuff for developing a Ribir user interface. So we place examples in it, there aren't other special reasons.

**Comment by zoechi** [comment]
Thanks. I haven't seen the `--example` flag before.
